tools/prediction_with_research_report/prediction_with_research_report.py

```diff
@@ -0,0 +1,163 @@
+   - "p_no": Probability that the market questions outcome will be `No`.
+   - "confidence": Indicating the confidence in the estimated probabilities you provided ranging from 0 (lowest confidence) to 1 (maximum confidence). Confidence can be calculated based on the quality and quantity of data used for the estimation.
+   - "info_utility": Utility of the information provided in "ADDITIONAL_INFORMATION" to help you make the probability estimation ranging from 0 (lowest utility) to 1 (maximum utility).
+* The sum of "p_yes" and "p_no" must equal 1.
+* Output only the JSON object in your response. Do not include any other contents in your response.
+"""
+
+
+def run(**kwargs) -> Tuple[str, Optional[Dict[str, Any]]]:
+    """Run the task"""
+    tool = kwargs["tool"]
+    prompt = kwargs["prompt"]
+    temperature = kwargs.get("temperature", DEFAULT_OPENAI_SETTINGS["temperature"])
+    max_compl_tokens = kwargs.get(
+        "max_tokens", DEFAULT_OPENAI_SETTINGS["max_compl_tokens"]
+    )
+    
+    openai_api_key = kwargs["api_keys"]["openai"]
+    tavily_api_key = kwargs["api_keys"]["tavily"]
+    
+    initial_subqueries_limit = kwargs.get('initial_subqueries_limit', DEFAULT_RESEARCH_SETTINGS["initial_subqueries_limit"])
+    subqueries_limit = kwargs.get('subqueries_limit', DEFAULT_RESEARCH_SETTINGS["subqueries_limit"])
+    scrape_content_split_chunk_size = kwargs.get('scrape_content_split_chunk_size', DEFAULT_RESEARCH_SETTINGS["scrape_content_split_chunk_size"])
+    scrape_content_split_chunk_overlap = kwargs.get('scrape_content_split_chunk_overlap', DEFAULT_RESEARCH_SETTINGS["scrape_content_split_chunk_overlap"])
+    top_k_per_query = kwargs.get('top_k_per_query', DEFAULT_RESEARCH_SETTINGS["top_k_per_query"])
+    
+    if tool not in ALLOWED_TOOLS:
+        raise ValueError(f"TOOL {tool} is not supported.")
+    
+    engine = TOOL_TO_ENGINE[tool]
+    
+    (research_report, _) = research(
+        prompt,
+        openai_key=openai_api_key,
+        tavily_key=tavily_api_key,
+        model=engine,
+        kwargs={
+            "initial_subqueries_limit": initial_subqueries_limit,
+            "subqueries_limit": subqueries_limit,
+            "scrape_content_split_chunk_size": scrape_content_split_chunk_size,
+            "scrape_content_split_chunk_overlap": scrape_content_split_chunk_overlap,
+            "top_k_per_query": top_k_per_query,
+        }
+    )
+
+    if tool not in ALLOWED_TOOLS:
+        raise ValueError(f"Tool {tool} is not supported.")
+
+    engine = TOOL_TO_ENGINE[tool]
+
+    prediction_prompt = PREDICTION_PROMPT.format(
+        user_prompt=prompt, additional_information=research_report
+    )
+    moderation_result = openai.Moderation.create(prediction_prompt)
+    if moderation_result["results"][0]["flagged"]:
+        return "Moderation flagged the prompt as in violation of terms.", None, None
+    messages = [
+        {"role": "system", "content": "You are a helpful assistant."},
```

Comment:
Could this intro system message be better? Something like you are a prediction agent that will have to make a forecast, etc

Reply:
We should be able to give it a real strategic starting prompt that embodies the research agent and preloads it with the necessary pretrained knowledge, avoid the garbage in garbage out paradigm

Reply:
If not, remove this entirely